test/tests/serial/open_dkim.bats

```diff
@@ -1,465 +1,288 @@
-    -v "${PRIVATE_CONFIG}/with-domain/opendkim":/etc/opendkim \
-    "${IMAGE_NAME}" /bin/bash -c 'ls -1 /etc/opendkim/keys/domain1.tld/ | wc -l'
-
-  assert_success
-  assert_output 2
-
-  # check keys for domain2.tld
-  run docker run --rm \
-    -e LOG_LEVEL='trace' \
-    -v "${PRIVATE_CONFIG}/with-domain/opendkim":/etc/opendkim \
-    "${IMAGE_NAME}" /bin/bash -c 'ls -1 /etc/opendkim/keys/domain2.tld | wc -l'
-
-  assert_success
-  assert_output 2
-
-  # check keys for domain3.tld
-  run docker run --rm \
-    -e LOG_LEVEL='trace' \
-    -v "${PRIVATE_CONFIG}/with-domain/opendkim":/etc/opendkim \
-    "${IMAGE_NAME}" /bin/bash -c 'ls -1 /etc/opendkim/keys/domain3.tld | wc -l'
-
-  assert_success
-  assert_output 2
-
-  # check keys for domain4.tld
-  run docker run --rm \
-    -e LOG_LEVEL='trace' \
-    -v "${PRIVATE_CONFIG}/with-domain/opendkim":/etc/opendkim \
-    "${IMAGE_NAME}" /bin/bash -c 'ls -1 /etc/opendkim/keys/domain4.tld | wc -l'
-
-  assert_success
-  assert_output 2
-
-  # check presence of tables and TrustedHosts
-  run docker run --rm \
-    -e LOG_LEVEL='trace' \
-    -v "${PRIVATE_CONFIG}/with-domain/opendkim":/etc/opendkim \
-    "${IMAGE_NAME}" /bin/bash -c "ls -1 /etc/opendkim | grep -E 'KeyTable|SigningTable|TrustedHosts|keys' | wc -l"
+function __assert_logged_dkim_creation() {
+  __assert_output_for 'Log' "${@}"
+}
 
-  assert_success
-  assert_output 4
+function __assert_output_for() {
+  local ASSERT_FOR=${1}
+  local EXPECTED_DOMAIN=${2}
+  local EXPECTED_SELECTOR=${3:-'mail'}
+
+  case "${ASSERT_FOR}" in
+    ( 'KeyTable' )
+      # EXAMPLE: mail._domainkey.domain1.tld domain1.tld:mail:/etc/opendkim/keys/domain1.tld/mail.private
+      assert_output --partial "${EXPECTED_SELECTOR}._domainkey.${EXPECTED_DOMAIN} ${EXPECTED_DOMAIN}:${EXPECTED_SELECTOR}:/etc/opendkim/keys/${EXPECTED_DOMAIN}/${EXPECTED_SELECTOR}.private"
+      ;;
+    ( 'SigningTable' )
+      # EXAMPLE: *@domain1.tld mail._domainkey.domain1.tld
+      assert_output --partial "*@${EXPECTED_DOMAIN} ${EXPECTED_SELECTOR}._domainkey.${EXPECTED_DOMAIN}"
+      ;;
+    ( 'Log' )
+      assert_output --partial "Creating DKIM private key '/tmp/docker-mailserver/opendkim/keys/${EXPECTED_DOMAIN}/${EXPECTED_SELECTOR}.private'"
+      ;;
+  esac
+}
 
-  # check valid entries actually present in KeyTable
-  run docker run --rm \
-    -e LOG_LEVEL='trace' \
-    -v "${PRIVATE_CONFIG}/with-domain/opendkim":/etc/opendkim \
-    "${IMAGE_NAME}" /bin/bash -c \
-    "egrep 'domain1.tld|domain2.tld|domain3.tld|domain4.tld' /etc/opendkim/KeyTable | wc -l"
+function __assert_outputs_common_dkim_logs() {
+  refute_output --partial 'No entries found, no keys to make'
+  assert_output --partial 'Creating DKIM KeyTable'
+  assert_output --partial 'Creating DKIM SigningTable'
+  assert_output --partial 'Creating DKIM TrustedHosts'
+}
 
-  assert_success
-  assert_output 4
+function __should_support_creating_key_of_size() {
+  local EXPECTED_KEYSIZE=${1}
```

Comment:
This function may not receive arguments at all, hence this would be slightly better (if we ever choose to use `set -u`):

```suggestion
  local EXPECTED_KEYSIZE=${1:-}
```